src/modm/ui/menu/choice_menu_impl.hpp

```diff
@@ -11,43 +12,48 @@
  */
 // ----------------------------------------------------------------------------
 
-#include "choice_menu.hpp"
+#ifndef CHOICE_MENU_HPP
+#	error 	"Don't include this file directly, use choice_menu.hpp instead!"
+#endif
 
+#include "abstract_view.hpp"
 
-modm::ChoiceMenu::ChoiceMenu(modm::ViewStack* stack, uint8_t identifier) :
-	modm::AbstractMenu(stack, identifier),
+template<typename Allocator>
+modm::ChoiceMenu<Allocator>::ChoiceMenu(modm::ViewStack<Allocator>* stack, uint8_t identifier) :
+	modm::AbstractMenu<Allocator>(stack, identifier),
 	display_update_time(500),
 	timer(display_update_time),
 	buttonAction(false),
 	title(""),
 	homePosition(0),
 	position(0)
 {
-	this->maximalDrawnEntrys = (getViewStack()->getDisplay().getHeight()- 16) / 8 ;
+	this->maximalDrawnEntrys = (this->getViewStack()->getDisplay().getHeight()- 16) / 8 ;
 }
 
-modm::ChoiceMenu::ChoiceMenu(modm::ViewStack* stack, uint8_t identifier, const char* title) :
-	modm::AbstractMenu(stack, identifier),
+template<typename Allocator>
+modm::ChoiceMenu<Allocator>::ChoiceMenu(modm::ViewStack<Allocator>* stack, uint8_t identifier, const char* title) :
+	modm::AbstractMenu<Allocator>(stack, identifier),
 	display_update_time(500),
 	timer(display_update_time),
 	buttonAction(false),
 	title(title),
 	homePosition(0),
 	position(0)
 {
-	this->maximalDrawnEntrys = (getViewStack()->getDisplay().getHeight()- 16) / 8 ;
+	this->maximalDrawnEntrys = (this->getViewStack()->getDisplay().getHeight()- 16) / 8 ;
 }
 
-void
-modm::ChoiceMenu::addEntry(const char* text,  bool *valuePtr, bool defaultValue)
+template<typename Allocator> void
+modm::ChoiceMenu<Allocator>::addEntry(const char* text,  bool *valuePtr, bool defaultValue)
 {
-	static uint16_t availableSpace = (getViewStack()->getDisplay().getWidth()-16)/6-6;
+	static uint16_t availableSpace = (this->getViewStack()->getDisplay().getWidth()-16)/6-6;
```

Comment:
Not sure what the reasoning behind this math is or what `availableSpace` represents

Reply:
me neither lol...this is stuff that we didn't write